From 4d6095a234402a48efd98f9e9a31f702c35dca3b Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 11 May 2015 18:47:25 +0800
Subject: [PATCH] task_gui.py: fix exception while invoking _anyRepoEnabled

Invoking _anyRepoEnabled to test whether the enabled repository is available,
while no repo is available(such as do not add any network repo in PXE), an
exception from _anyRepoEnabled break anaconda.

We should use False to instead of exception in this situation.

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/iw/task_gui.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pyanaconda/iw/task_gui.py b/pyanaconda/iw/task_gui.py
index c2b5a98..5321cd4 100644
--- a/pyanaconda/iw/task_gui.py
+++ b/pyanaconda/iw/task_gui.py
@@ -595,6 +595,8 @@ class TaskWindow(InstallWindow):
     def _anyRepoEnabled(self):
         model = self.rs.get_model()
         iter = model.get_iter_first()
+        if not iter:
+            return False
 
         while True:
             if model.get_value(iter, 0):
-- 
1.9.1

